pkg/dockerregistry/server/util.go

```diff
@@ -207,7 +214,176 @@ func (g *cachedImageStreamGetter) get() (*imageapi.ImageStream, error) {
 	return is, nil
 }
 
-func (g *cachedImageStreamGetter) cacheImageStream(is *imageapi.ImageStream) {
+func (g *cachedImageStreamGetter) cacheImageStream(is *imageapiv1.ImageStream) {
 	context.GetLogger(g.ctx).Debugf("(*cachedImageStreamGetter).cacheImageStream: got image stream %s/%s", is.Namespace, is.Name)
 	g.cachedImageStream = is
 }
+
+// imageMetadataFromManifest is used only when creating the image stream mapping
+// in registry. In that case the image stream mapping contains image with the
+// manifest and we have to pupulate the the docker image metadata field.
+func imageMetadataFromManifest(image *imageapiv1.Image) error {
```

Comment:
@smarterclayton we moved this to registry as this function is only used by registry (when you create ISM and need to place the metadata in RawExtension as registry use external client).

Reply:
> @smarterclayton we moved this to registry as this function is only used by registry (when you create ISM and need to place the metadata in RawExtension as registry use external client).

Moved?  Is this something that our API server did before?  Does the API server still do it if it is missing?  Recall that during an upgrade, we run skewed apiserver to registry.

Reply:
Moved or copied?  I don't see a corresponding delete.

Reply:
@smarterclayton moved in terms of this PR. This helper lived in v1/helper.go before (in this PR). We realized that the only user of this function is registry itself when we need to convert the Image in ISM to JSON (so when we create the ISM.

@deads2k we did this before, but we had a conversions that convert the external/internal image and fill-in all the fields we required... I don't like this pattern TBH (that external vs. internal use JSON vs. regular fields). I believe this was to support multiple versions of manifest/docker config. @miminar ?